pkg/secretlock/local/local_secret_lock.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/google/tink/go/subtle/random"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// package local provides a local secret lock service. The user must create a master key and store it
+// in a local file or an environment variable prior to using this service.
+//
+// The user has the option to encrypt the master key using NewMasterLock(passphrase, hash func(), salt)
+//
+// The master key must be stored (encrypted with MasterLock or not encrypted) either in a file or in
+// an environment variable.
+//
+// The user can then call either:
+//		MasterKeyFromPath(path) or
+//		MasterKeyFromEnv(envPrefix, keyURI)
+// to get an io.Reader instance needed to read the master key and create a keys Lock service.
+//
+// To get the lock service, call:
+// 		NewService(reader, secLock)
+// where reader is the instance returned from one of the previous Init functions.
+// and secLock in the masterKey lock used to encrypt/decrypt the master key, if it's nil
+// then the masterKey content in reader will be used as-is without being decrypted. The keys however are always
+// encrypted using the masterKey.
+
+var logger = log.New("aries-framework/lock")
+
+// Lock is a secret lock service responsible for encrypting keys using a master key
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// MasterKeyFromPath creates a new instance of a local secret lock Reader for a master key stored in `path`.
+func MasterKeyFromPath(path string) (io.Reader, error) {
+	masterKeyFile, err := os.OpenFile(path, os.O_RDONLY, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		err = masterKeyFile.Close()
+		if err != nil {
+			logger.Warnf("failed to close file: %w", err)
+		}
+	}()
+
+	mkData := make([]byte, 1024)
```

Comment:
why 1024. also should be a const.

Reply:
a large enough array, it can be half this size as we won't have keys larger than 32 bytes, encrypted and encoded, it should around < 100 bytes

I'll make it a const

Reply:
RSA keys are definitely bigger. TODO I guess.

Reply:
HKDF and AEAD don't work with RSA keys directly.. smaller keys are derived then used

Reply:
added const and set it to 512